### PR TITLE
Use `log` crate instead of `eprintln` & remove some unwraps

### PR DESCRIPTION
**Pull request by bircni**
- I fixed the TODO to use the `log` crate instead of `eprintln`
- Set the rust-version in the `scripts/check.sh` to the same as egui is on
- I made xtask use anyhow to remove some unwraps 

* [x] I have followed the instructions in the PR template

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -249,12 +249,12 @@ useless_transmute = "warn"
 verbose_file_reads = "warn"
 wildcard_dependencies = "warn"
 zero_sized_map_values = "warn"
+print_stderr = "warn"
```
Comment (YgorSouza) [inline Cargo.toml:252]: This list is sorted alphabetically.

File: xtask/src/utils.rs [hunk 2]
```diff
@@ -30,16 +28,16 @@ pub fn ask_to_run(mut cmd: Command, ask: bool, reason: &str) -> Result<(), DynEr
...
         match buf.trim().to_lowercase().as_ref() {
             "" | "y" | "yes" => {}
-            "n" | "no" => return Err("Aborting as per your request".into()),
-            a => return Err(format!("Invalid answer `{a}`").into()),
+            "n" | "no" => anyhow::bail!("Aborting as per your request"),
+            a => anyhow::bail!(format!("Invalid answer `{a}`")),
         };
     } else {
-        eprintln!("Running `{cmd:?}` to {reason}.");
+        log::info!("Running `{cmd:?}` to {reason}.");
```
Comment (YgorSouza) [inline xtask/src/utils.rs:35]: Don't know if this makes sense. xtask is supposed to be just like a shell script, but in Rust. The fewer bells and whistles it has, the better.

Comment (bircni) [reply xtask/src/utils.rs]: you are right removed that and replaced it with println

**Comment by bircni** [comment]
@emilk Could you have a look at it?

File: examples/puffin_profiler/Cargo.toml [hunk 3]
```diff
@@ -27,3 +27,4 @@ env_logger = { version = "0.10", default-features = false, features = [
 ] }
 puffin = "0.19"
 puffin_http = "0.16"
+log = { workspace = true }
```
Comment (emilk) [inline examples/puffin_profiler/Cargo.toml:30]: nit: sort dependencies

File: examples/puffin_profiler/src/main.rs [hunk 4]
```diff
@@ -153,7 +153,7 @@ fn start_puffin_server() {
 
     match puffin_http::Server::new("127.0.0.1:8585") {
         Ok(puffin_server) => {
-            eprintln!("Run:  cargo install puffin_viewer && puffin_viewer --url 127.0.0.1:8585");
+            log::info!("Run:  cargo install puffin_viewer && puffin_viewer --url 127.0.0.1:8585");
```
Comment (emilk) [inline examples/puffin_profiler/src/main.rs:156]: This means the user won't see the message unless they have `RUST_LOG=info` env-var set.

Either revert, or add this to before the call to `env_logger::init`:

```rs
// Show log messages by default:
let rust_log = std::env::var("RUST_LOG").unwrap_or_else(|_| "info".to_owned());
std::env::set_var("RUST_LOG", rust_log);
```

**Review by emilk — APPROVED** [review]

**Pull request merged**